Fixes #18845: restores sort behavior for DeviceTable.name column

jnovinger opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #18845 
- Restores sort order using `Device.name` column on `DeviceTable.name` table column.

This was a regression introduced in bf836c9bc2 (in v4.2.5).

<!--
    Please include a summary of the proposed changes below.
-->

jnovinger committed:
Fixes #18845: restores sort behavior for DeviceTable.name column

mraerino commented on netbox/dcim/tables/devices.py:
```diff
@@ -145,7 +145,8 @@ class DeviceTable(TenancyColumnsMixin, ContactsColumnMixin, NetBoxTable):
         verbose_name=_('Name'),
         accessor=Accessor('label'),
```
i think this might also need to be changed?
```suggestion
        accessor=Accessor('name'),
```

alehaa replied:
Instead of switching the accessor to name, it should be removed entirely. It was added in bf836c9 to display a value other than `name`.

I believe removing this line and switching the template `DEVICE_LINK` from `value` to `record.label` might preserve displaying the label but fix sorting. Sorry I didn't catch this before.

jeremystretch commented on netbox/dcim/tables/devices.py:
```diff
@@ -145,7 +145,8 @@ class DeviceTable(TenancyColumnsMixin, ContactsColumnMixin, NetBoxTable):
         template_code=DEVICE_LINK,
-        linkify=True
+        linkify=True,
+        order_by=('name',)
```
This isn't needed if we correct the accessor.

jeremystretch commented on netbox/dcim/tables/devices.py:
```diff
@@ -145,7 +145,8 @@ class DeviceTable(TenancyColumnsMixin, ContactsColumnMixin, NetBoxTable):
         verbose_name=_('Name'),
         accessor=Accessor('label'),
```
As @mraerino points out, the accessor needs to be reverted to `name`.

jnovinger commented:
@mraerino @jeremystretch , thanks for pointing that out. I had interpreted the [previous change (that caused this behavior)](https://github.com/netbox-community/netbox/pull/18710) to be intentional with respect to the display value of the object's `Name` column value.

Reverting the `accessor` argument reverts the behavior of using `Device.label` value for the `Name` column. See screenshots below, first is with my change, the second is with the suggested changes to this PR. In the second, you can see that devices belonging to the Tenant: NC State, Site: MDF, Location: Row 1 lose their descriptive value in the `Name` column and simply become `"Unnamed Device"`.

Behavior based on this PR as filed:
![image](https://github.com/user-attachments/assets/d652e42c-3b1c-44d9-b5c3-033643f4ad4b)

Behavior with PR suggested changes:
![image](https://github.com/user-attachments/assets/6f22c389-f087-4bf5-9f9a-f829f12ac5b4)


I just want to clarify, this is the behavior we want--that is, these devices show up with `"Unnamed Device"` in the `Name` column?

alehaa commented:
> The "Unnamed device" label signifies that name is null for the device (which is permitted).

@jeremystretch the intention of #17357 was exactly to change this behavior and show the same display name as in the rack elevations view. I proposed a change to fix sorting but preserving #17357 in the comments above: https://github.com/netbox-community/netbox/pull/18861#discussion_r1990177112

jeremystretch commented:
The column accessor should be set to a concrete value in the table data. If you want the column to display something other than the accessor's value, you need to use TemplateColumn or override the columns' `render()`/`value()` methods.

However, be aware that rendering attributes of any related objects which have not been prefetched in the querysey may trigger additional queries.

jnovinger committed:
Remove accessor/order_by and modify DEVICE_LINK template

Thanks to @alehaa for the suggestion.

This also includes an additional `.select_related()` operation on
`DeviceListView.queryset` to avoid extra queries. Thanks to
@renatoalmeidaoliveira and @jeremystretch for pointing out the need for
this.

jnovinger commented on netbox/dcim/tables/devices.py:
```diff
@@ -143,9 +143,8 @@ class Meta(NetBoxTable.Meta):
-        accessor=Accessor('label'),
         template_code=DEVICE_LINK,
-        linkify=True
+        linkify=True,
```
Oops, forgot I added this when adding the previous `order_by` argument. I'm a monster!
